src/_lib/modules/word-cloud/word-cloud.component.ts

```diff
@@ -0,0 +1,433 @@
+        exportCanvas.height = this._canvas.height;
+
+        this._positionedWords.forEach(pw => {
+            const point: IPoint = { x: pw.canvasX, y: pw.canvasY };
+            this._setFontDetails(exportCtx, pw.fontSize);
+            this._drawWord(pw, point, exportCtx);
+        });
+
+        return exportCanvas.toDataURL('image/png');
+    }
+
+    downloadCanvas(filename: string): void {
+        const dataUrl = this.exportCanvas();
+        const link = document.createElement('a');
+
+        link.download = `${filename}.png`;
+        link.href = dataUrl;
+        link.click();
+    }
+
+    onResize() {
+        this._init();
+    }
+
+    private _init(): void {
+        this.config = this._getConfig();
+        this._truncatedWordsWithFontSize = this._getTruncatedWordsWithFontSize(this.words);
+        this._positionedWords = [];
+        this._drawCanvas();
+        this._centerPoint = { x: this._canvas.width / 2, y: this._canvas.height / 2 };
+
+        this._isFontLoaded().subscribe(() => {
+            this._positionWords();
+    
+            if (this.config.debugMode) {
+                console.info('Config', this.config);
+                this._drawSprial();
+            }
+
+            this._cdRef.detectChanges();
+        });
+    }
+
+    /**
+     * Calculates the font size for each word based on its weight and the range of weights in the list of words
+     * @param words List of words with their respective weights
+     * @returns A new list of words sorted by weight (largest to smallest) with their respective font sizes
+     */
+    private _getTruncatedWordsWithFontSize(words: T[]): IWordWithFontSize<T>[] {
+        const weights = words.map(w => w.weight);
+        const minWeight: number = Math.min(...weights);
+        const maxWeight: number = Math.max(...weights);
+
+        return words.map(word => {
+            return {
+                ...word,
+                fontSize: this._getFontSize(word.weight, minWeight, maxWeight),
+                truncatedValue: this._truncateWord(word.value)
+            };
+        }).sort((a, b) => b.weight - a.weight);
+    }
+
+    private _drawCanvas(): void {
+        this._canvas = this._canvas || this._renderer.createElement('canvas');
+        this._canvas.width = this._elRef.nativeElement.parentElement.clientWidth;
+        this._canvas.height = this._elRef.nativeElement.parentElement.clientHeight;
+        this._ctx = this._canvas.getContext('2d');
+
+        if (this.config.debugMode) {
+            this._renderer.appendChild(this._elRef.nativeElement, this._canvas);
+        }
+    }
+
+    /**
+     * This method is responsible for positioning the words onto a canvas by iterating through each word, finding a suitable location for
+     * it using a spiral algorithm, and checking if the word's bounding box intersects with any existing bounding boxes
+     */
+    private _positionWords(): void {
+        const t1 = performance.now();
```

Comment:
Should we check if debugMode is true before declaring this?

Reply:
I was thinking about it but then `t1` is in a new scope in the `if` so I'd need to `let t1` above the if and it becomes a bit messy. For that reason I left it to execute regardless of debugMode